locksmith/src/utils/receipts.ts

```diff
@@ -32,3 +37,127 @@ export const getAllReceipts = async ({
   }
   return receipts
 }
+
+export const getReceiptsZipName = (lockAddress: string, network: number) => {
+  return `receipts-${network}-${lockAddress}.zip`
+}
+
+export const zipReceiptsAndSendtos3 = async (
+  lockAddress: string,
+  network: number
+): Promise<boolean> => {
+  const subgraphservice = new SubgraphService()
+  const receipts = await subgraphservice.receipts(
+    {
+      where: {
+        lockAddress: lockAddress.toLowerCase().trim(),
+      },
+    },
+    {
+      networks: [network],
+    }
+  )
+
+  if (!receipts.length) {
+    return false
+  }
+
+  const web3Service = new Web3Service(networks)
+  const lock = await web3Service.getLock(lockAddress, network)
+
+  const createPDFBuffer = (data: any) => {
+    const amount = `${Number(data.amountTransferred) / Math.pow(10, lock.currencyDecimals)} ${lock.currencySymbol}`
+
+    return new Promise((resolve, reject) => {
+      const doc = new PDFDocument()
+      const buffers: Buffer[] = []
+
+      doc.on('data', buffers.push.bind(buffers))
+      doc.on('end', () => resolve(Buffer.concat(buffers)))
+      doc.on('error', reject)
+
+      doc.fontSize(14).text('Receipt Number')
+      doc.moveDown(0.5)
+
+      doc.font('Helvetica-Bold').text(`#${data.receiptNumber}`)
+      doc.font('Helvetica').fillColor('black')
+      doc.moveDown()
+
+      doc.text('Transaction Date')
+      doc.moveDown(0.5)
+      doc
+        .font('Helvetica-Bold')
+        .text(`${new Date(data.timestamp * 1000).toDateString()}`)
+      doc.font('Helvetica').fillColor('black')
+      doc.moveDown()
+
+      doc.text('Transaction Hash')
+      doc.moveDown(0.5)
+      doc.font('Helvetica-Bold').text(`${data.id}`)
+      doc.moveDown()
+
+      doc.fontSize(14).fillColor('#603DEB').text('Bill to')
+      doc.fillColor('black')
+      doc.moveDown(0.5)
+      doc.font('Helvetica')
+      doc.text(`Wallet: ${data.payer}`)
+      doc.moveDown()
+
+      doc.font('Helvetica-Bold').fillColor('#603DEB').text('Service Performed')
+      doc.font('Helvetica').fillColor('black')
+      doc.moveDown(0.5)
+      doc.text('NFT membership')
+      doc.moveDown()
+
+      doc.font('Helvetica-Bold').fillColor('#603DEB').text('Amount')
+      doc.font('Helvetica').fillColor('black')
+      doc.moveDown(0.5)
+      doc.text('TOTAL', { align: 'left', continued: true })
+      doc.font('Helvetica-Bold').text(amount, { align: 'right' })
+      doc.font('Helvetica')
+      doc.moveDown()
+
```

Comment:
wow! Indeed.
Actually, I wonder if we could not just generate a PDF _from_ the HTML rendering rather than create a custom PDF. That would be simpler and MUCH easier to maintain.